packages/aws-cdk-lib/aws-eks/lib/access-entry.ts

```diff
@@ -0,0 +1,340 @@
+        namespaces: options?.namespaces,
+      };
+    }
+    return new Import();
+  }
+  /**
+   * The scope of the access policy, which determines the level of access granted.
+   */
+  public readonly accessScope: AccessScope;
+
+  /**
+   * The access policy itself, which defines the specific permissions.
+   */
+  public readonly policy: string;
+
+  /**
+   * Constructs a new instance of the AccessPolicy class.
+   *
+   * @param {AccessPolicyProps} props - The properties for configuring the access policy.
+   */
+  constructor(props: AccessPolicyProps) {
+    this.accessScope = props.accessScope;
+    this.policy = props.policy.policyArn;
+  }
+}
+
+/**
+ * Represents the properties required to create an Amazon EKS access entry.
+ */
+export interface AccessEntryProps {
+  /**
+   * The name of the AccessEntry.
+   *
+   * @default - No access entry name is provided
+   */
+  readonly accessEntryName?: string;
+  /**
+   * The Amazon EKS cluster to which the access entry applies.
+   */
+  readonly cluster: ICluster;
+  /**
+   * The access policies that define the permissions and scope for the access entry.
+   */
+  readonly accessPolicies: IAccessPolicy[];
+  /**
+   * The Amazon Resource Name (ARN) of the principal (user or role) to associate the access entry with.
+   */
+  readonly principal: string;
+}
+
+/**
+ * Represents an access entry in an Amazon EKS cluster.
+ *
+ * An access entry defines the permissions and scope for a user or role to access an Amazon EKS cluster.
+ *
+ * @implements {IAccessEntry}
+ */
+export class AccessEntry extends Resource implements IAccessEntry {
+  /**
+   * Imports an `AccessEntry` from its attributes.
+   *
+   * @param scope - The parent construct.
+   * @param id - The ID of the imported construct.
+   * @param attrs - The attributes of the access entry to import.
+   * @returns The imported access entry.
+   */
+  public static fromAccessEntryAttributes(scope: Construct, id: string, attrs: AccessEntryAttributes): IAccessEntry {
+    class Import extends Resource implements IAccessEntry {
+      public readonly accessEntryName = attrs.accessEntryName;
+      public readonly accessEntryArn = attrs.accessEntryArn;
+
+    }
+    return new Import(scope, id);
+  }
+  /**
+   * The name of the access entry.
+   */
+  public readonly accessEntryName: string;
+  /**
+   * The Amazon Resource Name (ARN) of the access entry.
+   */
+  public readonly accessEntryArn: string;
+  private cluster: ICluster;
+  private principal: string;
+  private accessPolicies: IAccessPolicy[];
+
+  constructor(scope: Construct, id: string, props: AccessEntryProps ) {
+    super(scope, id);
+
+    this.cluster = props.cluster;
+    this.principal = props.principal;
+    this.accessPolicies = props.accessPolicies;
+
+    const resource = new CfnAccessEntry(this, 'Resource', {
+      clusterName: this.cluster.clusterName,
+      principalArn: this.principal,
```

Comment:
Are we going to allow users to customize `type` on access entry? Or we don't plan to support it on first launch but can be an added feature in the future?

Reply:
The `type`(eks.AccessScopeType) is per AccessPolicy which can be defined this way and attach to an AccessEntry.

```ts
const clusterScopedClusterAdminPolicy = AccessPolicy.fromAccessPolicyName('AmazonEKSClusterAdminPolicy', {
  accessScope: { type: eks.AccessScopeType.CLUSTER },
});
const namespaceScopedAdminPolicy = AccessPolicy.fromAccessPolicyName('AmazonEKSAdminPolicy', {
  accessScope: { type: eks.AccessScopeType.NAMESPACE, namespaces: ['foo'] },
});

new AccessEntry(stack, 'Entry', {
  cluster,
  accessPolicies: [
    clusterScopedClusterAdminPolicy,
    namespaceScopedAdminPolicy,
  ],
  principal: clusterAdminRole.roleArn,
});
```

We have an internal ticket(V1385204928) to check if all policies support both `Cluster` and `Namespaces` AccessScopeType.